flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptivebatch/forwardgroup/ForwardGroupComputeUtilTest.java

```diff
@@ -162,20 +171,35 @@ public void testOneInputSplitsIntoTwo() {
         v2.getProducedDataSets().get(0).getConsumer().setForward(true);
         v2.getProducedDataSets().get(1).getConsumer().setForward(true);
 
-        Set<ForwardRegion> regions = computePipelinedRegions(v1, v2, v3, v4);
+        Set<ForwardGroup> groups = computeForwardGroups(v1, v2, v3, v4);
 
-        checkRegionSize(regions, 2, 3, 1);
+        checkGroupSize(groups, 1, 3);
     }
 
-    private static Set<ForwardRegion> computePipelinedRegions(JobVertex... vertices) {
+    private static Set<ForwardGroup> computeForwardGroups(JobVertex... vertices) throws Exception {
+        Arrays.asList(vertices).forEach(vertex -> vertex.setInvokableClass(NoOpInvokable.class));
+        ExecutionGraph executionGraph = createDynamicGraph(vertices);
         return new HashSet<>(
-                ForwardRegionComputeUtil.computeForwardRegions(Arrays.asList(vertices)).values());
+                ForwardGroupComputeUtil.computeForwardGroups(
+                                Arrays.asList(vertices), executionGraph::getJobVertex)
+                        .values());
     }
 
-    private static void checkRegionSize(
-            Set<ForwardRegion> regions, int numOfRegions, int... sizes) {
-        assertEquals(numOfRegions, regions.size());
+    private static void checkGroupSize(Set<ForwardGroup> groups, int numOfGroups, int... sizes) {
+        assertEquals(numOfGroups, groups.size());
         containsInAnyOrder(
-                regions.stream().map(ForwardRegion::size).collect(Collectors.toList()), sizes);
+                groups.stream().map(ForwardGroup::size).collect(Collectors.toList()), sizes);
+    }
+
+    private static DefaultExecutionGraph createDynamicGraph(JobVertex... vertices)
```

Comment:
Looks to me the test does not require a dynamic graph?

Reply:
Yes, but the forward groups will only be used in AdaptiveBatchScheduler(dynamic graph)

Reply:
The `ForwardGroup` is only used in dyanmic graph, so I think using dynamic graph here is better.